### ISSUE TITLE
Yellow OS Installer defaults to IPv6, can this be changed to IPv4

### ISSUE THREAD

**Issue by andyblac** [home-assistant/operating-system#2738]
### Describe the issue you are experiencing

The OS Yellow installer defaults to IPv6, but the installer got an IP address but failed to get an DNS server or IP, so therefore it got stuck on the `Steady Yellow LED` issue., The only way I could get the installer to work is to use an old router that did not support IPv6. Can you change this so it defaults to IPv4, IMO IPv6 should not be required for most users, and if it is could be setup later in the HA UI.

### What operating system image do you use?

yellow (Home Assistant Yellow)

### What version of Home Assistant Operating System is installed?

6.1.21-v8

### Did you upgrade the Operating System.

Yes

### Steps to reproduce the issue

1. flash usb with installer image
2. boot the Yellow

### Anything in the Supervisor logs that might be useful for us?

```txt
could not get to any logs.
```

### Anything in the Host logs that might be useful for us?

```txt
could not get to any logs.
```

### System information

could not get to any logs.

**Comment by agners** [comment]
> but the installer got an IP address but failed to get an DNS server or IP

Hm, do you know why that happens?

FWIW, I do have a IPv6/IPv4 setup here, and the Installer does work in my environment. So this must be some environment related issue.

The Installer should run in a dual stack configuration: It should get both, an IPv6 and and IPv4 address. By default, the system will prefer to use IPv6 to reach the server. But if the IPv6 address is a global address and routing or similar is broken, then things could fail. Ideally I guess the Installer should fall back to IPv4 (using [Happy Eyeballs](https://en.wikipedia.org/wiki/Happy_Eyeballs) algorithm) in case the IPv6 setup is not working correctly.

**Comment by agners** [comment]
Hm, make use of `curl` in the Installer, [which supports Happy Eyeballs](https://daniel.haxx.se/blog/2020/03/16/curl-ootw-happy-eyeballs-timeout/). :thinking: 

> it got stuck on the Steady Yellow LED issue

Which Installer version did you use? The yellow LED was steadily blinking correct?

**Comment by andyblac** [comment]
Hi,

I followed this [guide](https://yellow.home-assistant.io/poe-nvme/) and choose `Home Assistant OS Installer for Yellow`

btw, the yellow did also get a IPv4 IP as I could ping it.

The way I found the issue, was flashing the image direct to the NVME, and booting, I then to could to the IPv4 IP with port 8123, 
<img width="550" alt="PNG image" src="https://github.com/home-assistant/operating-system/assets/77936/c02b36d7-791f-4654-b2a3-2f6c0f31bb39">
then got and could click on the blue dot. I dont have the full log from there but I did keep this

```
23-04-04 12:15:12 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.DNS_SERVER_IPV6_ERROR/ContextType.DNS_SERVER
```

If it helps.
